### PR TITLE
added code for wolfssl dependency in libcurl

### PR DESCRIPTION
**Pull request by rushikesh90**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
Hi @rushikesh90 Why did you change the default value for `with_openssl`? 

Please, sign the CLA and include your name in #4

**Comment by rushikesh90** [comment]
@uilianries I was testing the recipe with those options. Now I have reverted those changes.

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -128,6 +137,13 @@ def requirements(self):
                 pass
             else:
                 self.requires("openssl/1.1.1g")
+        if self.options.with_wolfssl:
+            if tools.is_apple_os(self.settings.os) and self.options.darwin_ssl:
+                pass
```
Comment (dheater) [inline recipes/libcurl/all/conanfile.py:142]: Shouldn't having `darwin_ssl` (or `with_winssl`) **and** `with_wolfssl` raise a `ConanInvalidConfiguration` as it does in the changes above?

**Comment by dheater** [comment]
Looks like CI is failing for reasons not related to this recipe.

```
ERROR: Missing binary: automake/1.16.2:8f0570538ddd703bfc5400385a2fca9aa4a2b671
ERROR: Missing binary: libtool/2.4.6:1c19d2cb2281d70e5828ac1b9d3a4ed44671d717
automake/1.16.2: WARN: Can't find a 'automake/1.16.2' package for the specified settings, options and dependencies:
- Settings: os=Linux
- Options: 
- Dependencies: autoconf/2.69
- Requirements: autoconf/2.Y.Z
- Package ID: 8f0570538ddd703bfc5400385a2fca9aa4a2b671

ERROR: Missing prebuilt package for 'automake/1.16.2', 'libtool/2.4.6'
Try to build from sources with "--build=automake --build=libtool"
Or read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package"
```

Suggestions on how to resolve?

**Comment by danimtb** [comment]
This was affected by https://github.com/conan-io/conan-center-index/issues/2196, I'll rerun the CI nos as it should be already fixed

**Comment by danimtb** [comment]
I see `No space left on device`. Probably an issue in our Windows agents, sorry for the inconvenience

**Comment by danimtb** [comment]
@rushikesh90 I think something wrong happened with the diff. I think it is probably related to the merge

**Comment by rushikesh90** [comment]
@danimtb Can you tell me what it is?

**Comment by danimtb** [comment]
@rushikesh90 sorry, it seems unrelated to these changes. I have relaunched the build

**Comment by danimtb** [comment]
Anyway, I see you still have changes unrelated to libcurl, please clean the diff. Otherwise, this pull request could not be merged

**Pull request closed without merging**